Improve the layout of the "load game" page

lmoureaux opened the pull request:
This PR changes the layout of the "load game" screen to jump around less. There are still huge improvements to be made but I think it's a good start.

The new layout looks like this:
![image](https://user-images.githubusercontent.com/22327575/181930175-49022f41-1248-478f-a5fe-9a2097526b55.png)


After #1165.
Related to #115.

jwrober commented:
So with the default NightStalker theme, the dialog overlaps the LongTurn logo as shown above and here.

![image](https://user-images.githubusercontent.com/23037938/181937211-ff9fe25d-5714-4632-9697-5db67b0d1e57.png)
![image](https://user-images.githubusercontent.com/23037938/181937217-d3a5b7b0-bfc1-45c2-ac2f-91a49e77e565.png)

Looks unprofessional. I would suggest increasing the size the the dialog so the image shows uncovered in the lower right corner. Since the other shipped themes don't have an image there it should be fine.

Also if you open the dialog, click cancel and then load it again you will briefly see the new game dialog and then it is replaced by the load game dialog.

Lastly, if you open the dialog, pick a saved game to show the preview, click cancel and then load saved game again, the preview of the game you clicked on the first try still shows, but the save game file in the table is not selected. I think it should be clean every time.

lmoureaux commented:
> Also if you open the dialog, click cancel and then load it again you will briefly see the new game dialog and then it is replaced by the load game dialog.

I don't think this one is new.

jwrober commented:
> > Also if you open the dialog, click cancel and then load it again you will briefly see the new game dialog and then it is replaced by the load game dialog.
> 
> I don't think this one is new.

I can recreate in beta.3, so yea not new. But if can be fixed here while the code is being worked on would be good.

lmoureaux commented:
> > > Also if you open the dialog, click cancel and then load it again you will briefly see the new game dialog and then it is replaced by the load game dialog.
> > 
> > 
> > I don't think this one is new.
> 
> I can recreate in beta.3, so yea not new. But if can be fixed here while the code is being worked on would be good.

Looks like a timing-dependent issue, it doesn't happen very often for me.

lmoureaux commented:
Should have addressed most concerns. The LT logo will still be partially covered at small window sizes, but now it should at least be fine when the window is big enough.

jwrober reviewed (approved):
Looks good to me.

lmoureaux commented:
OS X CI: https://gitlab.freedesktop.org/ is down

pranavtifr reviewed (approved):
Looks good to me. Can merge after rebasing.
